tests/integration/test_service_manager.py

```diff
@@ -43,6 +46,24 @@ def test___grpc_service_without_discovery_service___start_service___service_host
     _validate_if_service_running_by_making_rpc(port_number)
 
 
+@pytest.mark.expect_discovery_service_stub_error(True)
```

Comment:
Is it necessary to use a marker for this?

Can direct test parameterization work for this test case? https://docs.pytest.org/en/7.1.x/how-to/fixtures.html#override-a-fixture-with-direct-test-parametrization

Reply:
Thanks for the approach! Direct fixture parameterization seems to be easy approach. I have added a new fixture `expect_discovery_service_error_stub` which returns a boolean `False` as default, thus choosing `FakeDiscoveryServiceStub` as default. For that one test case which needs `FakeDiscoveryServiceStubError`, overridden the default value with `True`.